Functional.Tests/Results/ResultFactoryTests.cs

```diff
@@ -336,5 +339,471 @@ public void WhereDelegateFalse()
 				.AssertFailure()
 				.Should()
 				.Be("abc");
+
+		public class WhenMerge2
+		{
+			[Fact]
+			public void ReturnsSuccessWhenAllSuccess()
+			{
+				var result1 = Result.Success<int, string>(1);
+				var result2 = Result.Success<int, string>(2);
+
+				Result.Merge(result1, result2)
+					.AssertSuccess()
+					.Should()
+					.Be((1, 2));
+			}
+
+			[Theory]
+			[ResultMerge2ReturnsFailureArrangement]
+			public void ReturnsFailureWhenAtLeastOneFailure(Result<int, string>[] resultCollection)
+			{
+				Result.Merge(
+						resultCollection[0],
+						resultCollection[1])
+					.AssertFailure()
+					.Should()
+					.Match(errorCollection
+						=> errorCollection.Count() == resultCollection.Count(r => !r.IsSuccess())
+						   && resultCollection.Where(r => !r.IsSuccess()).Select(r => r.Failure()).WhereSome().All(errorCollection.Contains));
+			}
+
+			#region Arrangements
+
+			private class ResultMerge2ReturnsFailureArrangement : AutoDataAttribute
+			{
+				public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+				{
+					for (int i = 0; i < (1 << 2) - 1; ++i)
+					{
+						var result1 = Result.Create((i & 1) == 1, () => 13, () => "error 1");
+						var result2 = Result.Create(((i >> 1) & 1) == 1, () => 17, () => "error 2");
+
+						yield return new object[] { new[] { result1, result2 } };
+					}
+				}
+			}
+
+			#endregion
+		}
+
+		public class WhenMerge3
+		{
+			[Fact]
+			public void ReturnsSuccessWhenAllSuccess()
+			{
+				var result1 = Result.Success<int, string>(1);
+				var result2 = Result.Success<int, string>(2);
+				var result3 = Result.Success<int, string>(3);
+				
+				Result.Merge(result1, result2, result3)
+					.AssertSuccess()
+					.Should()
+					.Be((1, 2, 3));
+			}
+
+			[Theory]
+			[ResultMerge3ReturnsFailureArrangement]
+			public void ReturnsFailureWhenAtLeastOneFailure(Result<int, string>[] resultCollection)
+			{
+				Result.Merge(
+						resultCollection[0],
+						resultCollection[1],
+						resultCollection[2])
+					.AssertFailure()
+					.Should()
+					.Match(errorCollection
+						=> errorCollection.Count() == resultCollection.Count(r => !r.IsSuccess())
+						   && resultCollection.Where(r => !r.IsSuccess()).Select(r => r.Failure()).WhereSome().All(errorCollection.Contains));
+			}
+
+			#region Arrangements
+
+			private class ResultMerge3ReturnsFailureArrangement : AutoDataAttribute
+			{
+				public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+				{
+					for (int i = 0; i < (1 << 3) - 1; ++i)
```

Comment:
Programmatically generate all 7 "at least one failure" cases.  The same methodology is applied for the other new tests.